src/labkey/query/Rows.spec.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as Ajax from '../Ajax'
+import {
+    deleteRows,
+    insertRows,
+    updateRows,
+} from './Rows'
+
+describe('deleteRows', () => {
+    it('should support original method signature', () => {
+        // Original method signature:
+        // deleteRows(schemaName, queryName, rows, success, failure)
+
+        // Arrange
+        const requestSpy = jest.spyOn(Ajax, 'request').mockImplementation();
+        const schemaName = 'SSS';
+        const queryName = 'QQQ';
+        const rows = [{ rowId: 1 }, { rowId: 2 }];
+
+        // Act
+        (deleteRows as any)(schemaName, queryName, rows, undefined, undefined);
+
+        // Assert
+        expect(requestSpy).toHaveBeenCalledWith(expect.objectContaining({
+            method: 'POST',
+            jsonData: expect.objectContaining({
+                queryName,
+                rows,
+                schemaName,
+            }),
+            url: '/query/deleteRows.api',
+        }));
+    });
+});
+
+describe('insertRows', () => {
+    it('should support original method signature', () => {
+        // Original method signature:
+        // insertRows(schemaName, queryName, rows, success, failure)
+
+        // Arrange
+        const requestSpy = jest.spyOn(Ajax, 'request').mockImplementation();
+        const schemaName = 'SSS';
+        const queryName = 'QQQ';
+        const rows = [{ rowId: 1 }, { rowId: 2 }];
+
+        // Act
+        (insertRows as any)(schemaName, queryName, rows, undefined, undefined);
+
+        // Assert
+        expect(requestSpy).toHaveBeenCalledWith(expect.objectContaining({
+            method: 'POST',
+            jsonData: expect.objectContaining({
+                queryName,
+                rows,
+                schemaName,
+            }),
+            url: '/query/insertRows.api',
+        }));
+    });
+});
+
+describe('updateRows', () => {
+    it('should support original method signature', () => {
+        // Original method signature:
+        // updateRows(schemaName, queryName, rows, success, failure)
+
+        // Arrange
+        const requestSpy = jest.spyOn(Ajax, 'request').mockImplementation();
+        const schemaName = 'SSS';
+        const queryName = 'QQQ';
+        const rows = [{ rowId: 1 }, { rowId: 2 }];
```

Comment:
these three const look to be duplicated in each test, could move them to the top of the file?

Reply:
I decided to be explicit for each test and keep them separate. It was intentional.